prow/plank/reconciler_test.go

```diff
@@ -306,13 +308,22 @@ func TestProwJobIndexer(t *testing.T) {
 		expected []string
 	}{
 		{
-			name:     "Matches all keys",
-			expected: []string{prowJobIndexKeyAll, prowJobIndexKeyPending, pendingTriggeredIndexKeyByName(pjName)},
+			name: "Matches all keys",
+			expected: []string{
+				prowJobIndexKeyAll,
+				prowJobIndexKeyPending,
+				pendingTriggeredIndexKeyByName(pjName),
+				pendingTriggeredIndexKeyByJobQueueName(pjJobQueue),
```

Comment:
nit: we should not use this method in tests - provide the literal here so someone who changes the semantics of the index knows they've done so

Reply:
This is actually fine, because the index only exists in Plank in memory, so if someone were to change its name or such, it would still work (as long as the correct name is referenced everywhere)

Reply:
Unless this is serialized somehow/needs to be forwards/backwards compatible using the same func to generate the key throughout production code and tests is best imho, because if it needs to be changed it doesn't require the tests to be updated. But it is not a big deal one way or the other

Reply:
I side with @alvaroaleman on this one, I think it's more practical to have this function called instead of having a separate literal. But if we wanted to have this literal after all I'd say this should be extracted to a separate task (to change it for the entire test suite) and is therefore outside of scope for this PR.